File: 1920-build-array-from-permutation/1920-build-array-from-permutation.c
int* buildArray(int* nums, int numsSize, int* returnSize){
    int *k;
    int i;
    k = (int*)malloc(sizeof(int) * numsSize);
    
    for(i = 0; i < numsSize ; ++i)
    {
        k[i] = nums[nums[i]];
    }
    *returnSize = numsSize;
    return k;
}

File: 1920-build-array-from-permutation/1920-build-array-from-permutation.java
//Follow-up: Can you solve it without using an extra space (i.e., O(1) memory)?

class Solution {
    public int[] buildArray(int[] nums) {
        int n = nums.length;
        
        for(int i = 0; i < n; i++)                          
            nums[i] = nums[i] + n * (nums[nums[i]] % n);     
        
        for(int i = 0; i < n; i++)
            nums[i] = nums[i] / n;
        
        return nums;
    }
}